lib/internal/crypto/random.js

```diff
@@ -119,6 +120,76 @@ function randomFill(buf, offset, size, cb) {
   _randomBytes(buf, offset, size, wrap);
 }
 
+// Largest integer we can read from a buffer.
+// e.g.: Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
+const RAND_MAX = 281474976710655;
+
+// Generates an integer in [min, max] range where both min and max are
+// inclusive.
+function randomInt(min, max, cb) {
+  // randomInt(max, cb)
+  // randomInt(max)
+  if (typeof max === 'function' || typeof max === 'undefined') {
+    cb = max;
+    max = min;
+    min = 0;
+  }
+  const isSync = typeof cb === 'undefined';
+  if (!isSync && typeof cb !== 'function') {
+    throw new ERR_INVALID_CALLBACK(cb);
+  }
+  if (!NumberIsSafeInteger(min)) {
+    throw new ERR_INVALID_ARG_TYPE('min', 'safe integer', min);
+  }
+  if (!NumberIsSafeInteger(max)) {
+    throw new ERR_INVALID_ARG_TYPE('max', 'safe integer', max);
+  }
+  if (!(max >= min)) {
+    throw new ERR_OUT_OF_RANGE('max', `>= ${min}`, max);
+  }
+
+  // First we generate a random int between [0..range]
+  const range = max - min + 1;
+
+  if (!(range <= RAND_MAX)) {
+    throw new ERR_OUT_OF_RANGE('max - min', `< ${RAND_MAX}`, range);
+  }
+
+  const excess = RAND_MAX % range;
+  const randLimit = RAND_MAX - excess;
+
+  if (isSync) {
+    // Sync API
+    while (true) {
+      const x = randomBytes(6).readUIntBE(0, 6);
```

Comment:
Performance suggestion: allocate a buffer once with `new FastBuffer(6)` and use `randomFillSync()` to fill it up, that avoids creating potentially many throwaway buffers.

(I recognize that on average we'll get lucky 50% or more on the first try so consider this a take-it-or-leave-it suggestion, not a must.)